frontend/src/container/SearchFilterMenu.jsx

```diff
@@ -0,0 +1,294 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { translate } from 'react-i18next'
+import PropTypes from 'prop-types'
+import { Icon } from 'tracim_frontend_lib'
+import DateFilter from '../component/Search/DateFilter.jsx'
+import ContentFacets from '../component/Search/ContentFacets.jsx'
+import CheckboxFilter from '../component/Search/CheckboxFilter.jsx'
+import {
+  ADVANCED_SEARCH_TYPE,
+  DATE_FILTER_ELEMENT,
+  SEARCH_FIELD_LIST
+} from '../util/helper.js'
+
+export class SearchFilterMenu extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showSearchFieldList: true,
+      createdRange: {
+        showFilter: true,
+        afterDateActive: false,
+        afterDate: '',
+        beforeDateActive: false,
+        beforeDate: ''
+      },
+      modifiedRange: {
+        showFilter: true,
+        afterDateActive: false,
+        afterDate: '',
+        beforeDateActive: false,
+        beforeDate: ''
+      }
+    }
+  }
+
+  handleOpenOrCloseSearchFields = () => {
+    this.setState(prev => ({ showSearchFieldList: !prev.showSearchFieldList }))
+  }
+
+  handleOpenOrCloseCreatedRange = () => {
+    this.setState(prev => ({
+      createdRange: {
+        ...prev.createdRange,
+        showFilter: !prev.createdRange.showFilter
+      }
+    }))
+  }
+
+  handleCheckboxCreatedRange = (type) => {
+    const { state } = this
+
+    if (type === DATE_FILTER_ELEMENT.AFTER) {
+      state.createdRange.afterDateActive
+        ? this.handleChangeCreatedDate('', DATE_FILTER_ELEMENT.AFTER)
+        : this.handleChangeCreatedDate(state.createdRange.afterDate, DATE_FILTER_ELEMENT.AFTER)
+      this.setState(prev => ({
+        createdRange: {
+          ...prev.createdRange,
+          afterDateActive: !prev.createdRange.afterDateActive
+        }
+      }))
+    } else {
+      state.createdRange.beforeDateActive
+        ? this.handleChangeCreatedDate('', DATE_FILTER_ELEMENT.BEFORE)
+        : this.handleChangeCreatedDate(state.createdRange.beforeDate, DATE_FILTER_ELEMENT.BEFORE)
+      this.setState(prev => ({
+        createdRange: {
+          ...prev.createdRange,
+          beforeDateActive: !prev.createdRange.beforeDateActive
+        }
+      }))
```

Comment:
This and the code in the `if` branch can be deduplicated away into a function.

you may use somethink like:

```js
      this.setState(prev => ({
        createdRange: {
          ...prev.createdRange,
          [key]: !prev.createdRange[key]
        }
      }))
```

with `key = 'beforeDateActive'`

Reply:
fix